### DISCUSSION TITLE
Running standalone model (food demand) model, encountering "Error: MAgPIE model run did not finish properly (fulldata.gdx is missing)

### DISCUSSION THREAD

**Discussion by CEEEJAYLEE** [magpiemodel/magpie#711] [Help]
Hi All,

I have also been trying to run the standalone food demand model, according to the instructions outlined in \magpie\standalone\HOWTO.md:

* **Running using R**: You can start a standalone model in the same manner as the main model by providing a cfg to the start_run function. The only difference is that cfg$model has to be set from "main.gms" to the wanted model (e.g. "standalone/demand_model.gms"). Everything else works as usual (output folder is created, full.gms is written, job is submitted to SLURM,...)

So, I created a new start script as per tutorials (https://magpiemodel.github.io/tutorials/t07-startscript), put the new one into the start folder (7: food, as shown below), and then unfortunately I encountered the problem whenever running the revised start script. 
Any idea what is going on?

Thanks very much for your help,
Best,
C.

# FOOD START SCRIPT CODE
### Load start_run(cfg) function which is needed to start MAgPIE runs
source("scripts/start_functions.R")

### Source the default config at this stage and then over-write it before starting the run.
source("config/default.cfg")

### short description of the actual run
cfg$title <- "Food"
## Change results folder name
cfg$results_folder <- "output/:title::date:"

### path to the submodel to be used relative to main model folder
cfg$model <- "standalone/demand_model.gms"   #def = "main.gms"

###start MAgPIE run
start_run(cfg = cfg, codeCheck = FALSE)

# BELOW IS THE ERROR UPON RUNNING THE SCRIPT
Main selection of MAgPIE start scripts
 ---------------------------------------------------------------------
 -> Scripts in this selection are actively managed and work out of  <-
 ->                             the box                             <-
 ---------------------------------------------------------------------
 1:           default | start run with default.cfg settings
 2:        check code | Checking code for consistency issues
 3:     download data | just download default.cfg input data
 4:         test runs | Test routine for standardized test runs
 5:          forestry | start run with Forestry (Endogenous)
 6: compilation check | download input and compile main.gms
 **7:              Food |**

Alternatively, choose a start script from another selection:
 8:             extra | Additional MAgPIE start scripts
 9:          projects | Project-specific MAgPIE start scripts
10:        deprecated | Deprecated scripts
**Choose a start script or folder: 7**

Choose submission type:
1: Direct execution
2: Background execution
3: Debug mode
Number: 1
Executing ./scripts/start/Food.R
2024-08-06 09:24:50.301086: try to acquire model lock...
2024-08-06 09:24:50.319524: acquired model lock in 0 secs.
Generating lockfile in 'output/Food_2024-08-06_09.24.50'... done.
Initializing run renv, see 'output/Food_2024-08-06_09.24.50/log_renv.txt'...done.
2024-08-06 09:26:58.928353: unlocked model.
Saving timePrepareStart and timePrepareEnd to runstatistics.rda

Starting MAgPIE...
[1] 2
Error: MAgPIE model run did not finish properly (fulldata.gdx is missing). Please check full.lst for further information!
Execution halted
2024-08-06 09:27:11.918409: unlocked model.
There were 44 warnings (use warnings() to see them)
PS D:\magpie>

**Comment by tscheypidi** [comment]
It's been a while that we tested to run the food demand model standalone. The `full.lst` might give you some hints what went wrong. It could be that the `demand_model.gms` is not compatible anymore to recent changes in the model and might require some updates.

**Reply by CEEEJAYLEE** [reply]
Thanks. Would it be possible to extract the results from the demand model and associated dietary intake pattern changes from the final result? If yes, do you mind sharing some information with me?

Cheers,
C

> It's been a while that we tested to run the food demand model standalone. The `full.lst` might give you some hints what went wrong. It could be that the `demand_model.gms` is not compatible anymore to recent changes in the model and might require some updates.

**Reply by tscheypidi** [reply]
you can always use the model results of a full run and extract all values either using a function from the [magpie4](https://github.com/pik-piam/magpie4/) package or directly targeting specific parameter or variable values using `readGDX` from the [gdx2 package](https://github.com/pik-piam/gdx2), but I can't currently support you beyond that.

**Reply by CEEEJAYLEE** [reply]
Thanks for your suggestion. Will look into that
